ground/src/main/java/com/google/android/ground/ui/surveyselector/SurveySelectorFragment.kt

```diff
@@ -43,18 +46,35 @@ class SurveySelectorFragment : AbstractFragment(), BackPressListener {
     super.onCreate(savedInstanceState)
     viewModel = getViewModel(SurveySelectorViewModel::class.java)
     adapter = SurveyListAdapter(viewModel, this)
-    // TODO(#2081): Merge the survey list flow and survey list state flow into a single stream.
-    lifecycleScope.launch { viewModel.getSurveyList().collect { adapter.updateData(it) } }
-    lifecycleScope.launch {
-      viewModel.surveyListState.collect { state -> state?.let { handleSurveyListState(it) } }
-    }
-    lifecycleScope.launch {
-      viewModel.errorFlow.collect {
-        Toast.makeText(requireContext(), it, Toast.LENGTH_SHORT).show()
+    lifecycleScope.launch { viewModel.uiState.filterNotNull().collect { updateUI(it) } }
```

Comment:
Nit: Not sure if Kotlin follows the Java naming guidelines 100%, but if that's the case `UI` should written as `Ui` instead.

Reply:
done